internal/iam/repository_user.go

```diff
@@ -306,6 +311,26 @@ func (r *Repository) LookupUserWithLogin(ctx context.Context, accountId string,
 	return obtainedUser, nil
 }
 
+// allowUserAutoVivify determines if a user can be autovivified based on the account's scope
+func (r *Repository) allowUserAutoVivify(ctx context.Context, acct *authAccount) (bool, error) {
+	const op = "iam.(Repository).allowUserAutoVivify"
+	if acct == nil {
+		return false, errors.New(errors.InvalidParameter, op, "missing account")
+	}
+	acctScope := allocScope()
+	acctScope.PublicId = acct.ScopeId
+	err := r.reader.LookupByPublicId(context.Background(), &acctScope)
+	if err != nil {
+		return false, errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("unable to lookup account's scope %s", acct.ScopeId)))
+	}
+	switch {
```

Comment:
This switch is fine but you could also just simplify it to `return acct.AuthMethodId == acctScope.PrimaryAuthMethodId, nil` if you like.

Reply:
ty!